Objects/complexobject.c

```diff
@@ -526,7 +526,10 @@ complex_pow(PyObject *v, PyObject *w, PyObject *z)
     }
     errno = 0;
     exponent = b;
-    int_exponent = (long)exponent.real;
+    if ((double)LONG_MIN <= exponent.real && exponent.real < (double)LONG_MAX)
```

Comment:
Just as a comment: the reason I suggested `-(double)LONG_MIN` instead of `(double)LONG_MAX` is that (a) it's easier to reason about, since `LONG_MIN` is (almost certainly) a power of 2 and so there's no error introduced in the long-to-double conversion, and (b) it makes a difference in the case of 32-bit longs: if `exponent.real` has value exactly equal to `2**31 - 1` then it can safely be converted to a `long`, but the `if` condition excludes it. So we miss one particular integer case here. It doesn't matter much - there's no undefined behaviour either way, but I do think `-(double)LONG_MIN` is a bit cleaner here.

Reply:
FWIW I find `-(double)LONG_MIN` looking odd but the explanation why is very clear. Since it's strictly more correct (even if the difference is academic), I'd suggesting adopting it with a comment essentially copying what Mark wrote above. It'll be a gem to find for future code readers. Think of the children!

Reply:
Actually, looking at the code a bit more, I think we should stop doing this altogether. The code is taking 'w' and converting it to a complex, and then unpacking the complex to check if w is an integer value that fits in a C long. We have a much easier way of checking for that...